src/EditorFeatures/Core/Implementation/InlineRename/InlineRenameSession.cs

```diff
@@ -118,9 +120,20 @@ internal partial class InlineRenameSession : ForegroundThreadAffinitizedObject,
             _baseSolution = _triggerDocument.Project.Solution;
             this.UndoManager = workspace.Services.GetService<IInlineRenameUndoManager>();
 
+            this._editAndContinueWorkspaceService = workspace.Services.GetService<IEditAndContinueWorkspaceService>();
+            this._editAndContinueWorkspaceService.BeforeDebuggingStateChanged += OnBeforeDebuggingStateChanged;
+
             InitializeOpenBuffers(triggerSpan);
         }
 
+        private void OnBeforeDebuggingStateChanged(object sender, DebuggingStateChangedEventArgs args)
```

Comment:
@wschae When the rename session is committed, we unsubscribe from OnBeforeDebuggingStateChanged, so this will not be called again. See Line 536 of this file below, in ```EndRenameSession```.